### ISSUE TITLE
Encourage simulation tools optionally be able to output changed models

### ISSUE THREAD

**Issue by jonrkarr** [SED-ML/sed-ml#116]
Presently implementations of model changes can only be indirectly tested by executing two simulations with the unmodified model and model changes and using reports to compare their results to infer that the model was correctly changed. This strategy is doable -- we've used it in our test suite. A simpler alternative is for simulation tools to output changed models and to directly evaluate the modified model.

More generally, this works in the direction of making the modification of models more modular from the execution of models (with the exception of SetValue in RepeatedTask). Separating modification and execution further opens up the possibility of modularly executing workflows by using different tools for different parts of the workflow, avoiding the need for every tool to be able to execute entire workflows.

**Comment by jonrkarr** [comment]
Lucian mentioned that tellurium can already do this. Our tool is architected such that this is possible. As I understand, VCell and iBioSim also already separate the modification and execution of models.

**Comment by matthiaskoenig** [comment]
I agree. We should add information on what tools should provide as additional outputs/files so that it becomes easier to reproduce and debug. This includes in my opinion things like
- models with applied changes
- DataGenerators as HDF5

**Comment by jonrkarr** [comment]
To take advantage of HDF5 (or similar formats) for facilitating downstream computations, data generators need to be grouped by their shape and ideally by their semantic meaning. This will become more important as data generators targets are enabled for things such as Jacobians. For example, putting a Jacobian in the same HDF5 data set as a set of time courses wouldn't be the best use of HDF5. Neither would putting each SED-ML data generator in a separate HDF5 data set. In my opinion, reports are closer to what's needed.
